### DISCUSSION TITLE
想问一下V1和V3有什么不同

### DISCUSSION THREAD

**Discussion by abceleung** [acheong08/ChatGPT#1272] [Q&A] [answered]
你好,

就我理解,V1是作者自己建立了一个伺服器,然后我们用家就透过这伺服器去用ChatGPT. 所以用家不用买Plus就可以用.

V3则是OpenAI API的Wrapper, 所以要付费才能用.

是这样没错吗? 我在研究其他用上revChatGPT的repo \(比如说这个[discord bot](https://github.com/Zero6992/chatGPT-discord-bot)\). 他们没详细解释不同版本有什么分别, 干脆过来问原作者算了

**Comment by CoolPlayLin** [comment]
V1是逆向工程，使用cookies或账号密码模拟用户登录来进行账号验证，有小概率被OpenAI封号的风险
V3是基于OpenAI官方的Python写的，使用API_KEY来进行身份验证，应该不会被封号

**Reply by abceleung** [reply]
谢回答。不过我看这repo的readme上面有一句
> > 3:35 PM - Rate limit at 5 requests / 10 seconds due to small server (I ran out of budget.)

不太明白，这是说V1会用到作者的伺服器吗？

另外，用V3需要付费吗? 谢谢

**Reply by CoolPlayLin** [reply, marked as answer]
V3的话是用的openai的库弄得，你可以看这个：https://platform.openai.com/docs/guides/chat
>不太明白，这是说V1会用到作者的伺服器吗？

对，但你也可以自建
